components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/datasource/configreader/ConfigReader.java

```diff
@@ -81,11 +87,40 @@
         return new HashMap<>();
     }
 
+    private static Map<String, Object> readAnalyticsSolutionsConfig() {
+        try {
+            return (Map<String, Object>) DataHolder.getInstance().getConfigProvider()
+                    .getConfigurationObject(ANALYTICS_SOLUTIONS_NAMESPACE);
+        } catch (Exception e) {
+            log.error("Failed to read analytics solutions namespace from deployment.yml ", e);
+        }
+        return new HashMap<>();
+    }
+
     public String getSolutionType() {
         Object solutionType = carbonConfigs.get(CARBON_CONFIGS_TYPE);
         return solutionType != null ? solutionType.toString() : DatasourceConstants.SP;
     }
 
+    public List<String> getSolutionTypesEnabled() {
+        List<String> solutionsList = new ArrayList();
+        if (analyticsSolutionsConfig != null) {
+            analyticsSolutionsConfig.forEach((k, v) -> {
+                        if ((Boolean) v) {
+                            if (k.equalsIgnoreCase(IS_ANALYTICS_ENABLED)) {
```

Comment:
Since this is a map, we can simply use a get, cant we? We know there only be 3 solutions